File: NaoBlocks.Web/Dtos/EditorSettings.cs
using Data = NaoBlocks.Core.Models;

namespace NaoBlocks.Web.Dtos
{
    public class EditorSettings
    {
        public bool CanConfigure { get; set; } = true;

        public Data.UserSettings? User { get; set; }

        public string? Toolbox { get; set; }

        public bool IsSystemInitialised { get; set; }
    }
}

File: Clients/Angular/src/app/data/editor-settings.ts
import { UserSettings } from './user-settings';

export class EditorSettings {
    user: UserSettings = new UserSettings();
    toolbox?: string;
    useEvents: boolean = false;
    isSystemInitialised: boolean = false;
    canConfigure: boolean = false;

    isLoaded: boolean = false;
}
